src/Bab/RabbitMq/VhostManager.php

```diff
@@ -180,14 +180,19 @@ private function createQueues(Configuration $config)
             }
 
             for ($i = 0; $i < count($retries); $i++) {
+                if ($i === 0) {
```

Comment:
yes I'll fix that. Don't you get the same problem with `$this->createDl()` L.138? And L.156 with the creation of the delay exchange?

The `createQueues` method is becoming very messy, and I find it weird to have a creation of exchange in both `createExchanges` and `createQueues`. Each method should have their own responsibilities. 
I don't know how you guys want to proceed, but I see two solutions on the top of my head:
- Declare all exchanges (including retry, delay...) in the `exchanges` config, then all exchanges will be declared clearly in `createExchanges`. This will need explicit declaration from the guy implementing the library, and more work on validation and so on.
- Repeat some of the work done in `createQueues` into `createExchanges` (check that a queue as a `retries` parameter for example) to find out which exchange we need to create before calling `createQueues`. It's not ideal, but at least it will bring much clarity.

What do you think?